src/sage/schemes/curves/zariski_vankampen.py

```diff
@@ -959,10 +1240,248 @@ def braid_monodromy(f):
+    blocks = []
+    for i in range(len(cuts) - 1):
+        block = [j for j in L1 if j > cuts[i] and j < cuts[i + 1]]
+        if len(block) > 0:
+            blocks.append(block)
+    shorts = []
+    for a in blocks:
+        A = B(a).super_summit_set()
+        res = None
+        for tau in A:
+            sg = (sg0 * B(a) / sg0).conjugating_braid(tau)
+            A1 = rightnormalform(sg)
+            par = A1[-1][0] % 2
+            A1 = [B(a) for a in A1[:-1]]
+            if len(A1) == 0:
+                b = B.one()
+            else:
+                b = prod(A1)
+            b1 = len(b.Tietze()) / (len(A1) + 1)
+            if res is None or b1 < res[3]:
+                res = [tau, A1, par, b1]
+        if res[2] == 1:
+            r0 = res[0].Tietze()
+            res[0] = B([i.sign() * (d - abs(i)) for i in r0])
+        res0 = res[:2]
+        shorts.append(res0)
+    return shorts
+
+
+@parallel
+def conjugate_positive_form_p(braid):
+    return conjugate_positive_form(braid)
+
+
+def braid2rels(L):
+    r"""
+    Return a minimal set of relations of the group ``F / [(b * F([j])) / F([j]) for j in (1..d)]``
+    where ``F = FreeGroup(d)`` and ``b`` is a conjugate of a positive braid . One starts
+    from the non-trivial relations determined by the positive braid and transform them in relations determined by ``b``.
+
+    INPUT:
+
+    - ``L`` -- a tuple whose first element is a positive braid and the second element is a list of permutation braids.
+
+    OUTPUT:
+
+    A list of Tietze words for a minimal set of relations of ``F / [(g * b) / g for g in F.gens()]``.
+
+    EXAMPLES::
+
+        sage: from sage.schemes.curves.zariski_vankampen import braid2rels
+        sage: B.<s0, s1, s2> = BraidGroup(4)
+        sage: L = ((s1*s0)^2, [s2])
+        sage: braid2rels(L)
+        [(4, 1, -2, -1), (2, -4, -2, 1)]
+    """
+    br = L[0]
+    L1 = L[1]
+    B = br.parent()
+    d = B.strands()
+    F = FreeGroup(d)
+    T = br.Tietze()
+    T1 = set(T)
+    m = len(T1) + 1
+    k = min(T1) - 1
+    B0 = BraidGroup(m)
+    F0 = FreeGroup(m)
+    br0 = B0([_-k for _ in T])
+    br0_left = leftnormalform(br0)
+    q, r = ZZ(br0_left[0][0]).quo_rem(2)
+    br1 = B0.delta()**r * B0(prod(B0(_) for _ in br0_left[1:]))
+    cox = prod(F0.gens())
+    U0 = [cox**q * (f0 * br1) / cox**q / f0 for f0 in F0.gens()[:-1]]
+    U = [tuple(sign(k1)*(abs(k1) + k) for k1 in _.Tietze()) for _ in U0]
+    pasos = [B.one()] + [_ for _ in reversed(L1)]
+    for C in pasos:
+        U = [(F(a) * C.inverse()).Tietze() for a in U]
+        ga = F / U
+        P = ga.gap().PresentationFpGroup()
+        dic = P.TzOptions().sage()
+        dic['protected'] = d
+        dic['printLevel'] = 0
+        P.SetTzOptions(dic)
+        P.TzGoGo()
+        P.TzGoGo()
+        gb = wrap_FpGroup(P.FpGroupPresentation())
+        U = [_.Tietze() for _ in gb.relations()]
+    return U
+
+
+@parallel
+def braid2rels_p(L):
+    return braid2rels(L)
```

Comment:
no doctest

Reply:
It is only used to paralleize `braid2rels`; I split since if not it did not appear in the documentation